### PR TITLE
hw-4.2b-solution

### PR DESCRIPTION
**Pull request by SviatoslavExpert**

### CODE REVIEW

File: src/main/java/core/basesyntax/Palindrome.java [hunk 1]
```diff
@@ -21,6 +21,15 @@
...
      */
     public boolean isPalindrome(String text) {
-        return false;
+        text = text.replaceAll("[^\\p{L}\\p{Nd}]+", "").toLowerCase();
+        StringBuilder sb = new StringBuilder(text);
+        String reversedText = sb.reverse().toString().toLowerCase();
+        int count = 0;
+        for (int i = 0; i < text.length(); i++) {
+            if (reversedText.charAt(i) == text.charAt(i)) {
+                count++;
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Palindrome.java]: @SviatoslavExpert you can simplify this)

Comment (boroda4436) [reply src/main/java/core/basesyntax/Palindrome.java]: you may require some algorithm knowlegde

Comment (boroda4436) [reply src/main/java/core/basesyntax/Palindrome.java]: but do not spend on this task too much time. Try to spent not more then 1 hour

Comment (SviatoslavExpert) [reply src/main/java/core/basesyntax/Palindrome.java]: Fixed. Hope this improvement will work out.

File: src/main/java/core/basesyntax/Palindrome.java [hunk 2]
```diff
@@ -21,6 +21,9 @@
      * <p>Результат: true</p>
      */
     public boolean isPalindrome(String text) {
-        return false;
+        text = text.replaceAll("[^\\p{L}\\p{Nd}]+", "").toLowerCase();
+        StringBuilder sb = new StringBuilder(text);
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Palindrome.java]: `sb` isn't very good name

Comment (SviatoslavExpert) [reply src/main/java/core/basesyntax/Palindrome.java]: sb changed to stringBuilder

**Review by Sofasmile — APPROVED** [review]

**Pull request closed without merging**